skimage/_shared/filters.py

```diff
@@ -9,13 +9,11 @@
 import numpy as np
 from scipy import ndimage as ndi
 
-from .._shared import utils
 from .._shared.utils import _supported_float_type, convert_to_float, warn
 
 
-@utils.deprecate_multichannel_kwarg(multichannel_position=5)
 def gaussian(image, sigma=1, output=None, mode='nearest', cval=0,
-             multichannel=None, preserve_range=False, truncate=4.0, *,
+             preserve_range=False, truncate=4.0, *,
              channel_axis=None):
```

Comment:
I'd like to point out

https://github.com/scikit-image/scikit-image/blob/8ae85b2998c05f2db8cab35b2fa0c293509d099d/skimage/_shared/filters.py#L107-L112

I am pretty sure that the logic makes it impossible to indicate to the function to treat an image of shape `(M, N, 3)` as a grayscale image as `channel_axis=None` is replaced with `channel_axis=-1`. The warning text implies, that in the absence of an explicit multichannel, the image is treated as RGB in some cases. However, with the switch to `channel_axis` we can no longer replicate this behavior. `channel_axis=None` means "no RGB" as opposed to `multichannel=None` which means "not set". 

Pretty sure this was broken since the `deprecate_multichannel_kwarg` was applied as well. A that one translates `multichannel=False` into `channel_axis=None`...

My proposal is to remove this (broken) implicit RGB assumption if `(M, N, 3)`.

Reply:
Oops! The warning message is problematic too; it should have been updated ever since `multichannel` became deprecated, so the user isn't invited to set a deprecated kwarg...

Reply:
We need another None-proxy, e.g.:

```python
class NoChannelAxis:
		pass

def gaussian(..., channel_axis=NoChannelAxis):
	if channel_axis is NoChannelAxis:
	    ...
			
# User can do:
gaussian(..., channel_axis=None)
```

Reply:
See a5421d5ff8936b7a25dfd1768881ac4aca78e02b. Patching the `__repr__` with a metaclass was actually a lot of fun, although it probably shouldn't be. :sweat_smile: 

I think the patch addresses the following cases:

1. User neither passed `multichannel` nor `channel_axis` and relied on automatic detection of color channel for `(M, N, 3)`. Since v0.19 we silently broke his workflow. This patch recovers the original behavior for one release and notifies the user.
2. User continued using `multichannel=False`. Since v0.19 that was replaced with `channel_axis=None`. In case the shape was `(M, N, 3)` we broke his code otherwise he was fine. This user is notified as well because `multichannel` is no longer available.
3. User read the misleading RuntimeWarning, and nevertheless started using `channel_axis=None` for a grayscale image. The warning continued. In case the shape was `(M, N, 3)` we broke his code otherwise he was fine. This patch recovers the original behavior but does not notify him.

All other cases using grayscale images shouldn't have been impacted.

Reply:
@stefanv @lagru What is the status of this? It would be good to get this merged soon.

Reply:
Are you asking about the status of this specific comment thread? It's been addressed in a5421d5ff8936b7a25dfd1768881ac4aca78e02b. 

Concerning the status of this PR, I think I addressed everything I found at the time. The only remaining suggestion was to add `.. versionchanged::` directives everywhere. But that should not hold up this PR in my opinion.

Reply:
Thank you for patching up this behavior @lagru, it looks great.